lib/pat.c

```diff
@@ -5833,18 +5901,34 @@ grn_pat_node_compare_by_key(const grn_id id1, const grn_id id2, void *arg)
   return 0;
 }
 
+static inline void
+pat_key_move(grn_ctx *ctx,
+             grn_pat *pat,
+             pat_node *node,
+             uint8_t *key,
+             uint32_t key_size,
+             uint32_t new_curr_key)
```

Comment:
```suggestion
             uint32_t new_position)
```

引数の順番は`new_position`, `key`, `key_size`がいいかな。`KEY_AT()`もそうだし。